tests/test_fields.py

```diff
@@ -505,6 +505,38 @@ async def name(self):  # noqa: D102
     assert await page.name == "namex"
 
 
+def test_field_processors_default() -> None:
+    def proc1(s):
+        return s + "x"
+
+    @attrs.define
+    class Page(ItemPage):
+        default_processors = [str.strip, proc1]
```

Comment:
Hm, I think default processors should be field-specific, i.e. it should be something like

```py
default_processors = {"name": [str.strip]}
```

I'm not sure about having a default processor for everything, because different fields have different data types.
Also, processors for different fields are likely different.

Reply:
> So that redefining a field in a subclass retains the out of the parent field?

Yes, I think that was the idea. Probably you don't even need to define the fields in the base class. 

This way you can create a base class with default processing logic for each field. We can't use e.g. attrs converters in item classes for that, as it makes page.field != item.field, but that's a useful feature, so we need another approach.

Reply:
> if most of your fields are strings

In zyte-common-items's Product 10/26 fields are not strings. So, while technically most fields are strings, there are many non-string fields; also, you'd want different default processing for the string fields anyways.

Reply:
Well, that's not the idea I was implementing. If we decide that we don't need to reimplement this `poe` feature this can be closed.

Reply:
How would you call `default_processors = {"name": [str.strip]}` attribute? I wonder if we should do something like

```py
default_processors = {
    "name": [str.strip],
    "*": [..],
}
```

Reply:
though with such dictionary it's a bit hard to add/change things in a subclass

Reply:
> So, while technically most fields are strings, there are many non-string fields; also, you'd want different default processing for the string fields anyways.

If we add the following field to this test page object:

```python
@field
def num(self):
    return 2
```

then calling `page.num` would result in `TypeError: descriptor 'strip' for 'str' objects doesn't apply to a 'int' object` for this test.

But I think in this case, it would be the user's responsibility to take into account the field type when declaring their field processors. For example, they could use `lambda x: x.strip if isinstance(x, str)` instead of `str.strip`.